### DISCUSSION TITLE
How to generate an API gateway from /openapi.json?

### DISCUSSION THREAD

**Discussion by igorgue** [fastapi/fastapi#7991] [Questions] [answered]
**Description**

I wonder if there's another project that can consume an openapi specification (that FastAPI :1st_place_medal: generates for us) and creates an API Proxy Gateway to pass requests to our my unsecured FastAPI microservice.

We currently do this manually, and it's a pain to me to see my framework generating this for me already and having to write code to match that with our JavaScript proxy.

**Additional context**

I got a microservice in FastAPI, we use Insomnia, and I see they are able to import a openapi.json file, that microservice gets requests from a proxy microservice that's more exposed online.

**Comment by JakeSummers** [comment]
Checkout https://github.com/swagger-api/swagger-codegen

**Comment by tiangolo** [comment, marked as answer]
Thanks for the help @JakeSummers !

@igorgue you might also want to check the Security docs, to add security (authentication, authorization) to your API: https://fastapi.tiangolo.com/tutorial/security/

**Comment by igorgue** [comment]
Thank you guys!!!
